app/models/group_user.rb

```diff
@@ -51,7 +53,7 @@ def self.update_first_unread_pm(last_seen, limit: 10_000)
           WHERE t.deleted_at IS NULL
           AND t.archetype = :archetype
           AND tu.last_read_post_number < CASE
-                                         WHEN u.admin OR u.moderator
+                                         WHEN u.admin OR u.moderator OR gu2.group_id IN (:whisperers_group_ids)
```

Comment:
I would actually do a conditional here. No need for an extra filter when no groups are configured.